graph/telemetry/istio/appender/util.go

```diff
@@ -11,20 +11,21 @@ import (
 	"github.com/kiali/kiali/graph"
 	"github.com/kiali/kiali/log"
 	"github.com/kiali/kiali/prometheus/internalmetrics"
+	"strings"
 )
 
 // package-private util functions (used by multiple files)
 
-func promQuery(query string, queryTime time.Time, api prom_v1.API, a graph.Appender) model.Vector {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
+func promQuery(query string, queryTime time.Time, ctx context.Context, api prom_v1.API, a graph.Appender) model.Vector {
```

Comment:
Note, that it's a more or less convention to put the context parameter in the first position of the signature.
I did for the k8s ones but on this util method I didn't want to churn too much the existing code, so I put it together with the API one.

Reply:
It's just a utility, if you feel the need to change it further go ahead.